3.0.0

matt-filion opened the pull request:
Proposal for 3.0.0 version of logger which is significantly lighter, as it removes Bunyan as a dependency. Also looks to add in some niceties around filtering out sensitive content, being more extensible and some additional decorators.
This is all in addition to miscellaneous bug fixes, and DRASTICALLY more unit testing.

matt-filion committed:
Initial 3.0.0 implementation.

Better handling of promises in decorators.
New decorator for duration.
Removed bunyan as default logger and added console handling.
JSON output for default logging.
Removed default logger.

matt-filion committed:
Duration messaging cleanup.

matt-filion committed:
Logger name cleanup for decorators.

matt-filion committed:
Unit testing and coverage.

matt-filion committed:
Test cases passing for current functionality.

matt-filion committed:
First draft of filters added

With an implementation for credit cards and keyword matching on objects passed in.

matt-filion committed:
Default filters have unit tests verifying them.

Also added verification on ConsoleLogger that filters will not modify the origin objects.

matt-filion committed:
Documentation

Also, Update paths to specific module. ES6 on compile.

matt-filion committed:
Module structure was off, couldnt be imported

Seems to be working as expected now.
Added some rules to tsconfig.json and did some cleanup as a result.

matt-filion committed:
Doc update, remove unused script command.

tiffin-filion commented on README.md:
```diff
@@ -1,46 +1,177 @@
+# Simple lightweight logging.
 
-Simple and lightweight logging.
+This logging frameowrk aims to be a good enough logging framework where being lightweight is an asset. For example FaaS situations like AWS Lambda, Azure Functions, , Google Functions, etc. There are a lot out there. A common problem is cold start, which bundle size has an impact.
```
```suggestion
This logging framework aims to be a good enough logging framework where being lightweight is an asset. For example FaaS situations like AWS Lambda, Azure Functions, Google Functions, etc. There are a lot out there. A common problem is cold start, where bundle size has an impact.
```

tiffin-filion commented on README.md:
```diff
@@ -1,46 +1,177 @@
-1. Log levels are integers 0 = trace, 3 = debug, 5 = info (default), 7 = warn, 10 = error
-1. For multiple values, separate each definition with a semi-colon ';'.
-1. To set the level for a specific group or namespace use the name first and the log level second.
+So this decorator will tell you the amount of time it takes for your function to execute. If this is an async function, it will be the amount of time util the promise resolves, not the amount of time to return the promise.
```
```suggestion
This decorator will tell you the amount of time it takes for your function to execute. If this is an async function, it will be the amount of time until the promise resolves, not the amount of time to return the promise.
```

tiffin-filion commented on README.md:
```diff
@@ -1,46 +1,177 @@
+
+Example: `X.duration.doASync: 0.366ms`
+
+But wait, thats not JSON! Yea, err'ing on the side of using out of the box behavior instead of build this out. Its easy to create memory leaks with this kind of behavior. When I do it, I want to take the time to do it right.
```
```suggestion
But wait, that's not JSON! Yea, erring on the side of using out of the box behavior instead of build this out. It's easy to create memory leaks with this kind of behavior. When I do it, I want to take the time to do it right.
```

tiffin-filion commented on README.md:
```diff
@@ -1,46 +1,177 @@
+
+But wait, thats not JSON! Yea, err'ing on the side of using out of the box behavior instead of build this out. Its easy to create memory leaks with this kind of behavior. When I do it, I want to take the time to do it right.
+
+## Level's
```
```suggestion
## Levels
```

tiffin-filion commented on README.md:
```diff
@@ -1,46 +1,177 @@
+
+## Level's
+
+Pretty standard here, highest level is fatal, lowest (most detailed) is trace. Lowest to highest they are all listed below.
```
```suggestion
Pretty standard here, highest level is fatal, lowest (most detailed) is trace. Levels listed below from lowest to highest:
```

tiffin-filion commented on README.md:
```diff
@@ -1,46 +1,177 @@
+* error `isError()` `error(...params)`
+* fatal `isFatal()` `fatal(...params)`
+
+For the default logger, each of tehse levels will use a different output method to take advantage of some of the built in functionality of `console`. So trace, debug and info are all reported using `console.log`, warn is reported using `console.warn` and then error and fatal are reported using `console.error`.
```
```suggestion
For the default logger, each of these levels will use a different output method to take advantage of some of the built in functionality of `console`. So trace, debug and info are all reported using `console.log`, warn is reported using `console.warn` and then error and fatal are reported using `console.error`.
```

tiffin-filion reviewed (changes requested):
Added suggestions you can accept/reject to fix spelling errors and grammar.

matt-filion committed:
Update README.md

Co-Authored-By: Tiffin Filion <39063170+tiffin-filion@users.noreply.github.com>

matt-filion committed:
Update README.md

Co-Authored-By: Tiffin Filion <39063170+tiffin-filion@users.noreply.github.com>

matt-filion committed:
Update README.md

Co-Authored-By: Tiffin Filion <39063170+tiffin-filion@users.noreply.github.com>

matt-filion committed:
Update README.md

Co-Authored-By: Tiffin Filion <39063170+tiffin-filion@users.noreply.github.com>

matt-filion committed:
Update README.md

Co-Authored-By: Tiffin Filion <39063170+tiffin-filion@users.noreply.github.com>

matt-filion committed:
Update README.md

Co-Authored-By: Tiffin Filion <39063170+tiffin-filion@users.noreply.github.com>

matt-filion committed:
Merge branch 'master' into 3.0.0

matt-filion committed:
3.0.3

Added adornments and updated ReadMe to reflect current state of capabilities.

matt-filion committed:
3.0.3

Clean up the logic for how log levels are found in the environment. Added overrides for environment values to 'force' a log level.

matt-filion committed:
3.0.3

Build configuration fix to output test case coverage.

jeff812 committed:
Update README.md

Fixing typos
